sunkit_instruments/goes_xrs/goes_chianti_tem.py

```diff
@@ -0,0 +1,212 @@
+from astropy import units as u 
+from astropy.io import fits
+from astropy.time import Time
+from sunpy import timeseries as ts 
+from sunpy.time import parse_time
+from sunpy.data import manager
+from scipy import interpolate
+import numpy as np 
+import pandas as pd
+from sunpy.util.exceptions import warn_user
+
+__all__ = ["goes_calculate_temperature_em", "_goes_chianti_temp_em", "_manage_goesr_detectors"]
+
+
+def goes_calculate_temperature_em(goes_ts, abundance="coronal", remove_scaling=False):
+    """
+    This calculates the temperature and emission measure from the GOES/XRS flux ratios.
+
+    Parameters
+    ----------
+    goes_ts : `~sunpy.timeseries.XRSTimeSeries`
+        The GOES XRS timeseries containing the data of both the xrsa and xrsb channels (in units of W/m**2).
+    abundance: `~str`, {`coronal` | `photospheric`}
+        Default 'coronal'. Which abundances to use for the calculation.
+    remove_scaling: `Boolean`
+        Checks whether to remove the SWPC scaling factors. Default False for "true" fluxes from the netcdf files. 
+        This is only an issue and will need to be used for the older FITS files for 8-15 XRS.
+
+    Returns
+    -------
+    `~sunpy.timeseries.GenericTimeSeries`
+        Conatins the temperature and emission measure calculated from the input `goes_ts` TimeSeries.
+
+    Example
+    -------
+    >>> goes_ts = ts.TimeSeries("sci_xrsf-l2-flx1s_g16_d20170910_v2-1-0.nc").truncate("2017-09-10 12:00", "2017-09-10 20:00")
+    >>> output = goes_calculate_temperature_em(goes_ts)
+
+    """
+    if not isinstance(goes_ts, ts.XRSTimeSeries):
+        raise TypeError(f"goes_ts must be a XRSTimeSeries object, not {type(goes_ts)}")
+
+    if goes_ts.observatory is None:
+        raise ValueError(f"The GOES satellite number is not found in the goes_ts")
+    else:        
+        satellite_number = int(goes_ts.observatory.split("-")[-1])
+        if (satellite_number <1) or (satellite_number>17):
+            raise ValueError(f"The GOES satellite number has to be between 1 and 17, {satellite_number} is given")
+
+    # check if GOES-R and whether the primary detector values are given
+    if (satellite_number>=16):  
+        if "primary_detector_a" in goes_ts.columns:
+            output = _manage_goesr_detectors(goes_ts, satellite_number, abundance=abundance)
+        else:
+            warn_user("No information about primary/secondary detectors in XRSTimeSeries, assuming primary for all")
+            output = _goes_chianti_temp_em(goes_ts, satellite_number, abundance=abundance)
+
+    # check if the older files are passed
+    elif ("Origin" in goes_ts.meta.metas[0]) and (goes_ts.meta.metas[0]["Origin"] == "SDAC/GSFC"):
```

Comment:
Does meta have a `.get` method?

Reply:
hmm do you mean do a goes_ts.meta.get("Origin")? it weirdly returns just an empty meta thing

Reply:
What about `goes_ts.meta.metas[0].get("Origin")`?